libs/remote_storage/src/simulate_failures.rs

```diff
@@ -0,0 +1,123 @@
+//! This module provides a wrapper around a real RemoteStorage implementation that
+//! causes some operations to fail. For testing purposes.
```

Comment:
> some

Let's describe how these operations fail and have.

Ideally, would be cool to have a unit test on that, but feels odd to test a test code.